text/0097-log-data-model.md

```diff
@@ -0,0 +1,988 @@
+</td>
+    <td>attributes, key=TBD</td>
+  </tr>
+  <tr>
+    <td>msg</td>
+    <td>string</td>
+    <td>Human readable message.</td>
+    <td>body</td>
+  </tr>
+  <tr>
+    <td>All other fields</td>
+    <td>any</td>
+    <td>Structured data.</td>
+    <td>attributes</td>
+  </tr>
+</table>
+
+### Apache HTTP Server access log
+
+<table>
+  <tr>
+    <td>Field</td>
+    <td>Type</td>
+    <td>Description</td>
+    <td>Maps to Unified Model Field</td>
+  </tr>
+  <tr>
+    <td>%t</td>
+    <td>timestamp</td>
+    <td>Time when an event occurred measured by the origin clock.</td>
+    <td>timestamp</td>
+  </tr>
+  <tr>
+    <td>%a</td>
+    <td>string</td>
+    <td>Client IP</td>
+    <td>attributes["net.peer.ip"]</td>
+  </tr>
+  <tr>
+    <td>%A</td>
+    <td>string</td>
+    <td>Server IP</td>
+    <td>attributes["net.host.ip"]</td>
+  </tr>
+  <tr>
+    <td>%h</td>
+    <td>string</td>
+    <td>Remote hostname. </td>
+    <td>attributes["net.peer.name"]</td>
+  </tr>
+  <tr>
+    <td>%m</td>
+    <td>string</td>
+    <td>The request method.</td>
+    <td>attributes["http.method"]</td>
+  </tr>
+  <tr>
+    <td>%v,%p,%U,%q</td>
+    <td>string</td>
+    <td>Multiple fields that can be composed into URL.</td>
+    <td>attributes["http.url"]</td>
+  </tr>
+  <tr>
+    <td>%>s</td>
+    <td>string</td>
+    <td>Response status.</td>
+    <td>attributes["http.status_code"]</td>
+  </tr>
+  <tr>
+    <td>All other fields</td>
+    <td>any</td>
+    <td>Structured data.</td>
+    <td>attributes, key=TBD</td>
+  </tr>
+</table>
+
+### CloudTrail Log Event
+
+<table>
+  <tr>
+    <td>Field</td>
+    <td>Type</td>
+    <td>Description</td>
+    <td>Maps to Unified Model Field</td>
+  </tr>
+  <tr>
+    <td>eventTime</td>
+    <td>string</td>
+    <td>The date and time the request was made, in coordinated universal time (UTC).</td>
+    <td>timestamp</td>
+  </tr>
+  <tr>
+    <td>eventSource</td>
+    <td>string</td>
+    <td>The service that the request was made to. This name is typically a short form of the service name without spaces plus .amazonaws.com.</td>
+    <td>resource["service.name"]?</td>
+  </tr>
+  <tr>
+    <td>awsRegion</td>
+    <td>string</td>
+    <td>The AWS region that the request was made to, such as us-east-2.</td>
+    <td>resource["cloud.region"]</td>
+  </tr>
+  <tr>
+    <td>sourceIPAddress</td>
+    <td>string</td>
+    <td>The IP address that the request was made from.</td>
+    <td>resource["net.peer.ip"] or resource["net.host.ip"]? TBD</td>
+  </tr>
+  <tr>
+    <td>errorCode</td>
+    <td>string</td>
+    <td>The AWS service error if the request returns an error.</td>
+    <td>short_description</td>
+  </tr>
+  <tr>
+    <td>errorMessage</td>
+    <td>string</td>
+    <td>If the request returns an error, the description of the error.</td>
+    <td>body</td>
+  </tr>
+  <tr>
+    <td>All other fields</td>
+    <td>*</td>
+    <td></td>
+    <td>attributes["cloudtrail.*"]</td>
+  </tr>
+</table>
+
```

Comment:
Hello,
I am Jake from the Google Cloud Logging team.  Could you add this mapping from Google Cloud Logging's logs into this universal data model. @otoolep for visibility.

Field | Type | Description | Maps to Unified Model Field
-- | -- | -- | --
timestamp | string | The time the event described by the log entry occurred. | timestamp
resource | MonitoredResource | The monitored resource that produced this log entry. | resource
log_name | string | The URL-encoded LOG_ID suffix of the log_name field identifies which log stream this entry belongs to. | shortname
json_payload | google.protobuf.Struct | The log entry payload, represented as a structure that is expressed as a JSON object. | body
proto_payload | google.protobuf.Any | The log entry payload, represented as a protocol buffer. | body
text_payload | string | The log entry payload, represented as a Unicode string (UTF-8). | body
severity | LogSeverity | The severity of the log entry. | severity
trace | string | The trace associated with the log entry, if any. | traceid
span_id | string | The span ID within the trace associated with the log entry. | spanid
labels | map<string,string> | A set of user-defined (key, value) data that provides additional information about the log entry. | attributes
All other fields | * |   | attributes["google.*"]

Reply:
Added, thanks!